code/game/objects/effects/effect_system/fluid_spread/effects_foam.dm

```diff
@@ -117,11 +118,15 @@
 		return FALSE
 	if(!istype(foaming))
 		return FALSE
+	if(foaming.foam_delay)
+		return
 
 	delta_time = min(delta_time SECONDS, lifetime)
 	var/fraction = (delta_time * MINIMUM_FOAM_DILUTION) / (initial(lifetime) * max(MINIMUM_FOAM_DILUTION, group.total_size))
-	reagents.reaction(foaming, TOUCH, fraction)
+	reagents.reaction(foaming, TOUCH|VAPOR, fraction)
 	lifetime -= delta_time
+	foaming.foam_delay = TRUE
+	addtimer(VARSET_CALLBACK(foaming, foam_delay, FALSE), 1 SECONDS)
```

Comment:
I feel like this should take ``delta_time`` into consideration

Reply:
I'm not sure I can find a solution that both takes delta_time into consideration, won't take disproportionally long if it happens on a longer or shorter tick than normal, and works across multiple sources of foam.

Reply:
Actually from what I can tell I'm pretty sure timers take lag into account so it should be fine to keep it as is.